kolibri_oidc_client_plugin/settings.py

```diff
@@ -12,10 +12,18 @@
 OIDC_RP_CLIENT_SECRET = os.environ.get("CLIENT_SECRET", "kolibri.app")
 OIDC_RP_SIGN_ALGO = "RS256"
 OIDC_AUTHENTICATION_CALLBACK_URL = "oidc_client:oidc_authentication_callback"
-OIDC_OP_AUTHORIZATION_ENDPOINT = "{}/authorize".format(OIDC_URL)
-OIDC_OP_JWKS_ENDPOINT = "{}/jwks".format(OIDC_URL)
-OIDC_OP_TOKEN_ENDPOINT = "{}/token".format(OIDC_URL)
-OIDC_OP_USER_ENDPOINT = "{}/userinfo".format(OIDC_URL)
+OIDC_OP_AUTHORIZATION_ENDPOINT = OPTIONS["OIDCClient"][
+    "AUTHORIZATION_ENDPOINT"
+] or "{}/authorize".format(OIDC_URL)
+OIDC_OP_JWKS_ENDPOINT = OPTIONS["OIDCClient"]["JWKS_URI"] or "{}/jwks".format(
+    OIDC_URL
+)
+OIDC_OP_TOKEN_ENDPOINT = OPTIONS["OIDCClient"]["TOKEN_ENDPOINT"] or "{}/token".format(
+    OIDC_URL
+)
+OIDC_OP_USER_ENDPOINT = OPTIONS["OIDCClient"]["USERINFO_ENDPOINT"] or "{}/userinfo".format(
+    OIDC_URL
+)
```

Comment:
Maybe a little cleaner to use `.get`? Also I would double-check `black` formatting

```suggestion
OIDC_OP_AUTHORIZATION_ENDPOINT = OPTIONS["OIDCClient"].get(
    "AUTHORIZATION_ENDPOINT", "{}/authorize".format(OIDC_URL)
)
OIDC_OP_JWKS_ENDPOINT = OPTIONS["OIDCClient"].get(
    "JWKS_URI", "{}/jwks".format(OIDC_URL)
)
OIDC_OP_TOKEN_ENDPOINT = OPTIONS["OIDCClient"].get(
    "TOKEN_ENDPOINT", "{}/token".format(OIDC_URL)
)
OIDC_OP_USER_ENDPOINT = OPTIONS["OIDCClient"].get(
    "USERINFO_ENDPOINT", "{}/userinfo".format(OIDC_URL)
)
```

Reply:
About black formatting: the code has passed through black, so this is its output :)
About using get: I considered it but I wanted to keep it consistent with the way these options are used in the rest of the Kolibri application. As it does not has any effect on the code I don't have any oposition, as you prefer.

Reply:
> About using get: I considered it but I wanted to keep it consistent with the way these options are used in the rest of the Kolibri application. As it does not has any effect on the code I don't have any opposition, as you prefer.


They're actually slightly different, and I'm not sure which one is more correct:

```
>>> test = {'key': ''}
>>> test['key'] or 'alternate'
'alternate'
>>> test.get('key', 'alternate')
''
```


generally though I prefer [`get(key[, default])`](https://docs.python.org/3/library/stdtypes.html#dict.get) because the intended behavior is called out explicitly by the function docs

> Return the value for `key` if `key` is in the dictionary, else `default`